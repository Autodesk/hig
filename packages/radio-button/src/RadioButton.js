import React, { Component } from "react";
import PropTypes from "prop-types";
import { CheckboxBehavior } from "@hig/checkbox";

import RadioButtonPresenter from "./presenters/RadioButtonPresenter";

export default class RadioButton extends Component {
  static propTypes = {
    /**
     * Checks the checkbox
     */
    checked: PropTypes.bool,
    /**
     * Initially checks the checkbox, but allows user action to change it
     */
    defaultChecked: PropTypes.bool,
    /**
     * Prevents user actions on the checkbox
     */
    disabled: PropTypes.bool,
    /**
     * Text identifying the field
     */
    label: PropTypes.string,
    /**
     * The name of the checkbox as submitted with a form
     */
    name: PropTypes.string,
    /**
     * Called when user moves focus from the field
     */
    onBlur: PropTypes.func,
    /**
     * Called when user clicks on the field
     */
    onClick: PropTypes.func,
    /**
     * Called when user changes the value of the field
     */
    onChange: PropTypes.func,
    /**
     * Called when user puts focus on the field
     */
    onFocus: PropTypes.func,
    /**
     * Marks the field as required, text shown to explain requirment
     */
    required: PropTypes.string,
    /**
     * Value submitted with a form if checked
     */
    value: PropTypes.string
  };

  render() {
    const {
      checked: controlledChecked,
      defaultChecked,
      disabled,
      label,
      name,
      onBlur,
      onChange,
      onClick,
      onFocus,
      required,
      value
    } = this.props;

    return (
      <CheckboxBehavior
        checked={controlledChecked}
        defaultChecked={defaultChecked}
        onChange={onChange}
        onClick={onClick}
      >
        {({ checked, handleClick }) => (
          <RadioButtonPresenter
            checked={checked}
            disabled={disabled}
            label={label}
            name={name}
            onBlur={onBlur}
            onClick={handleClick}
            onFocus={onFocus}
            required={required}
            value={value}
          />
        )}
      </CheckboxBehavior>
    );
  }
}
