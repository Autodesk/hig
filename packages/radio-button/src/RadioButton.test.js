import { mount } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import RadioButton from "./RadioButton";

describe("radio-button/RadioButton", () => {
  describe("integration", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(<RadioButton label="HIG RadioButton" />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("event handlers", () => {
    const subject = (props = {}) =>
      mount(<RadioButton title="RadioButton" {...props} />);
    let wrapper;

    const eventHandler = jest.fn();
    eventHandler.mockReset();

    describe("onBlur", () => {
      beforeEach(() => {
        wrapper = mount(<RadioButton onBlur={eventHandler} />);
      });

      it("is triggered on blur", () => {
        wrapper.find("input").simulate("blur");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onChange", () => {
      beforeEach(() => {
        wrapper = mount(<RadioButton onChange={eventHandler} />);
      });

      it("is triggered on change", () => {
        wrapper.find("input").simulate("change");

        expect(eventHandler).toHaveBeenCalled();
      });
    });

    describe("onFocus", () => {
      beforeEach(() => {
        wrapper = subject({ onFocus: eventHandler });
      });

      it("is triggered on focus", () => {
        wrapper.find("input").simulate("focus");
        expect(eventHandler).toBeCalled();
      });
    });
  });
});
