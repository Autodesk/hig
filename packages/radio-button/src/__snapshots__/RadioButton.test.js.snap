// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio-button/RadioButton integration renders correctly 1`] = `
<div
  className="hig__radio-button"
>
  <input
    checked={false}
    className="hig__radio-button__input"
    disabled={false}
    id="radio-button-1"
    name="radiobutton"
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    type="radio"
    value="value"
  />
  <span
    className="hig__radio-button__wrapper"
  >
    <span
      className="hig__radio-button__wrapper__button"
    />
    <span
      className="hig__radio-button__wrapper__button--disabled"
    />
  </span>
  <label
    className="hig__radio-button__label"
    htmlFor="radio-button-1"
  >
    HIG RadioButton
  </label>
</div>
`;
